docs/decision_record.md

```diff
@@ -6,6 +6,52 @@
 
 [TOC]
 
+## 2021-12-01 - Endpoint names and structure
+
+### Decision
+
+The endpoint names will be:
+
+- `GET /service-info` for GA4GH service info
+- `GET /collection/{digest}` for retrieving a sequence collection
+- `GET /comparison/{digest1}/{digest2}` for comparing two collections in the database
+- `POST /comparison/{digest1}/` for comparing one database collection to a local user-provided collection.
+
+The POST body for the local comparison is a "level 2" sequence collection, like this:
+
+```
+{
+  "lengths": [
+    "248956422",
+    "242193529",
+    "198295559"
+  ],
+  "names": [
+    "chr1",
+    "chr2",
+    "chr3"
+  ],
+  "sequences": [
+    "a004bc1b0bf05fc668cab6bbfd93d3eb",
+    "0ccf3a67666ac53f99fcad19768f2dde",
+    "bda7b228789169ae811dd8d676d517ca"
+  ]
+}
+```
+
+### Rationale
+
+We wanted to stick with the REST guideline of noun endpoints with GET that describe what you are retrieving. We believe a prefix, like `/seqcol/...` could be added by a service that implemented multiple specifications, but this kind of namespace it outside the scope of the specification itself. We considered doing `/{digest1}/compare/{digest2}` and that would have been fine. In the end we liked the symmetry of `/comparison` and `/collection` as parallel endpoints. For the retrieval endpoint we considered `/secol` or `/sequence-collection` or `/seqCol`, but wanted to keep structure parallel to the refget `/sequence` endpoint.
```

Comment:
One possible concern is `/service-info`. In refget this is located at `/sequence/service-info` but that's also because all other refget functions are under `sequence`. Here `service-info` could clash with others on the same server.

Reply:
Based on my interpretation from our reading of the GA4GH recommendations last week, refget is doing it incorrectly (probably because it predated the guidelines), and the seqcol way is the way it should happen.

If someone is putting 2 services on one server (say, mixing refget and seqcol), they would prefix them with a namespace. It doesn't make sense to build that namespace into the spec, since it lives outside the spec, which is self-contained within its namespace.

Reply:
See: https://github.com/ga4gh-discovery/ga4gh-service-info#how-do-i-describe-a-service-implementing-multiple-specifications